cfgov/unprocessed/js/apps/financial-well-being/fwb-results.js

```diff
@@ -1,72 +1,112 @@
 /* eslint no-extra-semi: "off" */
 'use strict';
 
-var Analytics = require( '../../modules/Analytics' );
-var Expandable = require( '../../organisms/Expandable' );
+const Analytics = require( '../../modules/Analytics' );
+const Expandable = require( '../../organisms/Expandable' );
+
+const expandableDom = document.querySelectorAll( '.content .o-expandable' );
+let expandable;
 
-var expandableDom = document.querySelectorAll( '.content .o-expandable' );
-var expandable;
 if ( expandableDom ) {
-  for ( var i = 0, len = expandableDom.length; i < len; i++ ) {
+  for ( let i = 0, len = expandableDom.length; i < len; i++ ) {
     expandable = new Expandable( expandableDom[i] );
     expandable.init();
   }
 }
 
+/**
+ * Initialize the results interactions
+ */
 function init() {
 
-  var toggleButtons = document.querySelectorAll(
+  const buttonsDom = document.querySelectorAll(
     '.comparison-chart_toggle-button'
   );
 
+  /**
+   * Changes the visibility of the results by category
+   * based on user input
+   * @param {string} category The category to display
+   */
   function switchComparisons( category ) {
-    var allCategories = document.querySelectorAll( '.comparison_data-point' );
-    var selectedButtonClass = 'comparison-chart_toggle-button__selected';
+    const allCategories = document.querySelectorAll( '.comparison_data-point' );
+    const showCategory = document.querySelectorAll(
+      '[class^="comparison_data-point ' + category + '"]'
+    );
+    const selectedButton = document.querySelector(
+      '[data-compare-by="' + category + '"]'
+    );
+    const selectedButtonClass = 'comparison-chart_toggle-button__selected';
+    const hiddenClass = 'u-hidden';
 
     // Hide all categories ...
     [].forEach.call( allCategories, function( el ) {
-      el.style.display = 'none';
+      el.classList.add( hiddenClass );
     } );
     // ... and deselect all toggle buttons ...
-    [].forEach.call( toggleButtons, function( el ) {
```

Comment:
It might be worth int to have a local forEach utility function.

Reply:
I like that. You mind if I push that to a future PR? This one is getting kinda unruly.